src/hyp3_isce2/burst.py

```diff
@@ -0,0 +1,238 @@
+    paths = [annotation_path, calibration_path, measurement_path]
+    for p in paths:
+        if not p.exists():
+            p.mkdir(parents=True)
+
+    et_args = {'encoding': 'UTF-8', 'xml_declaration': True}
+
+    ET.ElementTree(burst.annotation).write(annotation_path / burst.annotation_name, **et_args)
+    ET.ElementTree(burst.calibration).write(calibration_path / burst.calibration_name, **et_args)
+    ET.ElementTree(burst.noise).write(calibration_path / burst.noise_name, **et_args)
+    ET.ElementTree(burst.manifest).write(safe_path / 'manifest.safe', **et_args)
+
+    burst_params = BurstParams(burst.safe_name, burst.swath, burst.polarization, burst.burst_number)
+    download_burst(asf_session, burst_params, measurement_path / burst.measurement_name)
+
+    return safe_path
+
+
+# TODO currently only validated for descending orbits
+def get_region_of_interest(
+    poly1: geometry.Polygon, poly2: geometry.Polygon, is_ascending: bool = True
+) -> Tuple[float, float, float, float]:
+    bbox1 = geometry.box(*poly1.bounds)
+    bbox2 = geometry.box(*poly2.bounds)
+    intersection = bbox1.intersection(bbox2)
+    bounds = intersection.bounds
+
+    x, y = (0, 1) if is_ascending else (2, 1)
+    roi = geometry.Point(bounds[x], bounds[y]).buffer(0.005)
+    bounds = roi.bounds  # returns (minx, miny, maxx, maxy)
+    return bounds
+
+
+def get_asf_session() -> requests.Session:
+    # requests will automatically use the netrc file:
+    # https://requests.readthedocs.io/en/latest/user/authentication/#netrc-authentication
+    session = requests.Session()
+    payload = {
+        'response_type': 'code',
+        'client_id': 'BO_n7nTIlMljdvU6kRRB3g',
+        'redirect_uri': 'https://auth.asf.alaska.edu/login',
+    }
+    response = session.get('https://urs.earthdata.nasa.gov/oauth/authorize', params=payload)
+    response.raise_for_status()
+    return session
+
+
+def download_bursts(param_list: Iterator[BurstParams]) -> List[BurstMetadata]:
+    """Steps
+    For each burst:
+        1. Download metadata
+        2. Create BurstMetadata object
+        3. Create directory structure
+        4. Write metadata
+        5. Download and write geotiff
+    """
+    bursts = []
+    with get_asf_session() as asf_session:
+        for i, params in enumerate(param_list):
+            print(f'Creating SAFE {i+1}...')
+            metadata_xml = download_metadata(asf_session, params)
+            burst = BurstMetadata(metadata_xml, params)
+            spoof_safe(asf_session, burst)
+            bursts.append(burst)
+
+    print('SAFEs created!')
+
+    return bursts
+
+
+if __name__ == '__main__':
+    burst_params1 = BurstParams('S1A_IW_SLC__1SDV_20211229T231926_20211229T231953_041230_04E66A_3DBE', 'IW2', 'VV', 3)
+    burst_params2 = BurstParams('S1A_IW_SLC__1SDV_20220110T231926_20220110T231953_041405_04EC57_103E', 'IW2', 'VV', 3)
+    metadata = download_bursts([burst_params1, burst_params2])
```

Comment:
Does this just serve as a demo? Also, the `metadata` variable is never used.

Reply:
Yes this is just a demo. This script has utility outside of the plugin (for creating SAFE-style burst products so I wanted to provide an example. Happy to remove if you don't think it fits here though.

Reply:
I decided to remove it.